src/util/container.h

```diff
@@ -196,16 +196,17 @@ class MutexedQueue
 
 	MutexedQueue()
 	{
+		std::deque<T>();
 	}
 	bool empty()
 	{
 		JMutexAutoLock lock(m_mutex);
-		return (m_size.GetValue() == 0);
+		return (std::deque<T>::size() == 0);
```

Comment:
Since you're inheriting from `std::deque` you can just do `return size() == 0;`.